components/brave_wallet_ui/components/extension/encryption-key-panel/index.tsx

```diff
@@ -42,18 +43,19 @@ function EncryptionKeyPanel (props: Props) {
     accounts,
     selectedNetwork,
     encryptionKeyPayload,
+    decryptPayload,
     onProvideOrAllow,
     onCancel
   } = props
   const [isDecrypted, setIsDecrypted] = React.useState<boolean>(false)
 
   const foundAccountName = React.useMemo(() => {
-    return accounts.find((account) => account.address.toLowerCase() === encryptionKeyPayload.address.toLowerCase())?.name
-  }, [encryptionKeyPayload])
+    return accounts.find((account) => account.address.toLowerCase() === (encryptionKeyPayload || decryptPayload).address.toLowerCase())?.name
+  }, [encryptionKeyPayload, decryptPayload])
```

Comment:
We can utilize the `panelType` prop here
`const payloadAddress = panelType === 'request' ? encryptionKeyPayload.address : decryptPayload.address`